### PR TITLE
EntityManager methods for working with components based on name

### PR DESCRIPTION
**Pull request by mirrorcult**
See https://discord.com/channels/310555209753690112/560845886263918612/927682504956211250

This is used pretty rarely (I think some wizard spell does it in SS14, and status effects do it as well) but it's still pretty useful and reduces boilerplate a good amount in those cases. Used rarely enough that I didn't feel the need to add entitysys proxy methods for these

### CODE REVIEW

**Comment by Acruid** [comment]
> Random artifact should pick one of the components which will define how artifact will be activated

This is the terrible legacy way to do it, with the equivalent of `IActivate` interfaces.

> So I have CompA, CompB, CompC that should be added at runtime.

Needing a separate component for each interaction permutation is a bad idea.

You solution would work, but it's trying to solve a problem that is a bad idea in the first place. I bet the spells and status effects are doing the same things (component per spell/status effect in the strategy pattern), which is why content code would actually need a way to bind components to yaml using the component name.

> "Yeah, but my case is so unique, I don't think it will be widely used anywhere else"

https://en.wikipedia.org/wiki/Strategy_pattern, you will find this is many places, it's one of the basic 23 patterns.

Basically merging this is just going to reinforce bad code. This is going the opposite direction of https://github.com/space-wizards/RobustToolbox/pull/2389.

**Comment by mirrorcult** [comment]
binding to a component using a datafield and just taking the name in YAML would be nice but I don't really think we have a way to do that easily?

**Comment by mirrorcult** [comment]
that doesn't really solve everything though, statuseffects in content has to have some way to send which component is being dynamically added/removed over the wire and the name works for that. i guess that could use the netid instead of the name and then this isn't an issue?

**Comment by PaulRitter** [comment]
> binding to a component using a datafield and just taking the name in YAML would be nice but I don't really think we have a way to do that easily?

afaict componentmanager is already initialized at the time of serialization, so this would very much be possible, but what do you mean by binding? resolving it into a type? or resolving it into a componentdependency of sorts?

**Review by PaulRitter — CHANGES_REQUESTED** [review]
i agree with acruid. if anything, we should add a customtypeserializer for <Type, ValueDataNode> that'll resolve componentnames into a type.

if you want, i can quickly add that, shouldn't be too hard

**Comment by mirrorcult** [comment]
this is prolly a bad idea
